jedi/_compatibility.py

```diff
@@ -34,8 +34,32 @@ def close(self):
         del self.loader
 
 
-def find_module_py33(string, path=None):
-    loader = importlib.machinery.PathFinder.find_module(string, path)
+def find_module_py34(string, path=None):
+    implicit_namespace_pkg = False
+    spec = None
+    loader = None
+
+    if isinstance(path, ImplicitNSInfo):
+        old_dotted_path = path.name
+        path = path.paths
+    spec = importlib.machinery.PathFinder.find_spec(string, path)
+    if hasattr(spec, 'origin'):
+        origin = spec.origin
```

Comment:
The origin is needed to determine whether or not we are dealing with implicit namespace packages. If `origin == 'namespace'` we can safely assume we are dealing with implicit namespace packages.